src/audio/dai.c

```diff
@@ -293,6 +293,7 @@ static int dai_playback_params(struct comp_dev *dev)
 	config->src_width = comp_sample_bytes(dev);
 	config->dest_width = comp_sample_bytes(dev);
 	config->cyclic = 1;
+	config->irq_enabled = !pipeline_is_timer_driven(dev->pipeline);
```

Comment:
you added a function `pipeline_is_timer_driven()` in the previous patch. Here you initialise a parameter to `!pipeline_is_timer_driven()` and then below all your tests are again for `...->irq_enabled`. Maybe better negate the meaning of this parameter to either `.irq_disabled` or `.timer_based`